packages/opds-web-client/src/components/BorrowButton.tsx

```diff
@@ -1,12 +1,12 @@
 import * as React from "react";
 
-export interface BorrowButtonProps extends React.HTMLProps<any> {
+export interface BorrowButtonProps extends React.HTMLProps<{}> {
   borrow: () => Promise<any>;
 }
 
-export default class BorrowButton extends React.Component<BorrowButtonProps, void> {
+export default class BorrowButton extends React.Component<BorrowButtonProps, {}> {
   render() {
-    let props = Object.assign({}, this.props);
+    let props = JSON.parse(JSON.stringify(this.props));
```

Comment:
The previous version seems safer in case the props have something that can't be stringified. Like `props.children`, if it's more than just a string.

Reply:
I kept getting warnings about not being about to delete a property that is read-only. But the previous version is much safer and didn't know about the `props.children` case. Will change back.

Reply:
I'm having trouble using `Object.assign` because using it makes Typescript think that it's missing ~250 props that belong on an anchor element (like `focus`, `aria-selected`, etc). I think it's fine to use `JSON.parse(JSON.stringify(...))` here because the only props that are not needed are `borrow` and `children`.

This may also help remove any props that are not needed:
```
  const { borrow, children, ...rest } = this.props;
  let props = JSON.parse(JSON.stringify(rest));
```

But doing:
```
  const { borrow, children, ...rest } = this.props;
  let props = Object.assign({}, rest);
```

Still returns
```
...
            Type '{}' is missing the following properties from type 'HTMLButtonElement': autofocus, disabled, form, formAction, and 255 
```

This works but defeats the purpose: `let props = Object.assign<any, any>({}, this.props);`

Reply:
What do you think of doing `let props = Object.assign<HTMLButtonElement, any>({} as HTMLButtonElement, this.props);`?

Reply:
If you use the `...rest` version, I don't think you still need the `Object.assign` at all? I think it was making a copy so it could delete some props without deleting them from the original, but `rest` already accomplishes that.

Reply:
Just using `rest` without copying also throws the same error.

Reply:
It seems to work if you do `const { ref, borrow, children, ...rest } = this.props;` - I think the `ref` prop causes the error.

Reply:
Yes! Thanks, it is `ref` that causes the error. Will update the rest.